### ISSUE TITLE
Cannot build due to RUST dependencies 

### ISSUE THREAD

**Issue by heyong4725** [eclipse-zenoh/zenoh#41]
Cannot compile rust-master Zenoh code due to crates dependencies on a new git clone. Some of the .toml files have outdated dependencies. For example, async-sse is still on v3.0.0, while the latest version is v4.0.1. Currently async-sse v3.0.0 won't compile.

If you update the tide to latest 0.13.0 (which will use latest async-sse v4.0.1), then  plugins/zenoh-http/src/lib.rs won't compile.

$ cargo outdated
zenoh
================
Name        Project  Compat  Latest   Kind    Platform
----        -------  ------  ------   ----    --------
async-std   1.6.4    ---     1.6.5    Normal  ---
async-task  3.0.0    ---     Removed  Normal  ---

zenoh-protocol
================
Name        Project  Compat  Latest   Kind    Platform
----        -------  ------  ------   ----    --------
async-std   1.6.4    ---     1.6.5    Normal  ---
async-task  3.0.0    ---     Removed  Normal  ---

zenoh-util
================
Name        Project  Compat  Latest   Kind    Platform
----        -------  ------  ------   ----    --------
async-std   1.6.4    ---     1.6.5    Normal  ---
async-task  3.0.0    ---     Removed  Normal  ---

zenoh-router
================
Name        Project  Compat  Latest   Kind    Platform
----        -------  ------  ------   ----    --------
async-std   1.6.4    ---     1.6.5    Normal  ---
async-task  3.0.0    ---     Removed  Normal  ---

zplugin-example
================
Name        Project  Compat  Latest   Kind    Platform
----        -------  ------  ------   ----    --------
async-std   1.6.4    ---     1.6.5    Normal  ---
async-task  3.0.0    ---     Removed  Normal  ---

zplugin-http
================
Name              Project  Compat  Latest   Kind    Platform
----              -------  ------  ------   ----    --------
async-sse         3.0.0    ---     4.0.1    Normal  ---
async-std         1.6.4    ---     1.6.5    Normal  ---
async-task        3.0.0    ---     Removed  Normal  ---
route-recognizer  0.1.13   ---     0.2.0    Normal  ---
tide              0.11.0   ---     0.13.0   Normal  ---

zplugin_storages
================
Name        Project  Compat  Latest   Kind    Platform
----        -------  ------  ------   ----    --------
async-std   1.6.4    ---     1.6.5    Normal  ---
async-task  3.0.0    ---     Removed  Normal  ---

**Comment by heyong4725** [comment]
I made some code changes at my local clone: 

- bump tide crate to 0.13.0
- add #[derive(Clone)] before Session struct @ zenoh/src/net/session.rs

I am able to compile and run zenoh instance.

**Comment by JEnoch** [comment]
Fixed by commit 61a13ea.
Actually we don't want to have the zenoh::net::Session deriving Clone, because it would require more complex code to deal with the fact that the Session might be closed (if a Session is closed, its clones shall no longer be usable).
Keeping the Session non-Clone allows to have the close(self) operation consuming the ownership of self, and not care about possible clones.

That said, I fixed the compilation issue in zenoh-http using tide 0.13.0 by wrapping the Session in an Arc.
